torch/_dynamo/variables/torch.py

```diff
@@ -614,6 +614,14 @@ def fn_with_prim_types(x):
             return UserFunctionVariable(
                 torch.nn.init._calculate_correct_fan, **options
             ).call_function(tx, args, {})
+        elif (
+            self.value is torch.nested.nested_tensor
+            and kwargs.get("layout", torch.strided) == torch.strided
+        ) or self.value in (
+            torch._nested_tensor_from_mask,
+            torch._nested_from_padded,
+        ):
+            raise unimplemented("torch.compile does not support C++ NestedTensor")
```

Comment:
nit: reference strided layout instead of C++ NestedTensor in error messages, because I think that's the way we want to discriminate between them publicly.

similarly throughout